src/core/regimes.rkt

```diff
@@ -175,127 +175,131 @@
-  ;; We have num-candidates candidates, each of whom has error lists of length num-points.
-  ;; We keep track of the partial sums of the error lists so that we can easily find the cost of regions.
-  (define num-candidates (vector-length err-lsts-vec))
-  (define num-points (vector-length can-split-vec))
-  (define min-weight (fl num-points))
-
+  ;; Converting list of list to list of flvectors
+  ;; flvectors are used to remove pointer chasing
   (define (make-vec-psum lst) 
    (flvector-sums (list->flvector lst)))
-  (define flvec-psums (vector-map make-vec-psum err-lsts-vec))
+  (define flvec-psums (vector-map make-vec-psum (list->vector err-lsts)))
+
+  ;; Set up data needed for algorithm
+  (define number-of-alts (vector-length flvec-psums))
+  (define number-of-points (vector-length can-split-vec))
+  ;; min-weight is used as penalty to favor not adding split points
+  (define min-weight (fl number-of-points))
+  
+  ;; These 3 vectors are will contain the output data and be used for
+  ;; determining which alt is best for a given point
+  (define result-error-sums (make-flvector number-of-points))
+  (define result-alt-idxs (make-vector number-of-points))
+  (define result-prev-idxs (make-vector number-of-points))
+
+  ;; Temporary vector used to find the alt with least error for each point
+  (define vec-temp (make-flvector number-of-alts))
+  ;; TODO invert loops to match core algorithm data priority
+  (for ([point-idx (in-range number-of-points)])
+    ;; record the error for each candidate
+    (for ([alt-idx (range number-of-alts)])
+     (flvector-set! vec-temp alt-idx
+      (flvector-ref (vector-ref flvec-psums alt-idx) point-idx)))
+    ;; Find the min, no built in function to find smallest fl in vector
+    (define min-v (flvector-ref vec-temp 0))
+    (define min-idx 0)
+    (for ([val vec-temp] [idx (range number-of-alts)])
+      (cond [(< val min-v)
+             (set! min-idx idx)
+             (set! min-v val)]))
+    (flvector-set! result-error-sums point-idx (fl min-v))
+    (vector-set! result-alt-idxs point-idx min-idx)
+    (vector-set! result-prev-idxs point-idx number-of-points))
   
-  ;; Our intermediary data is a list of cse's,
-  ;; where each cse represents the optimal splitindices after however many passes
-  ;; if we only consider indices to the left of that cse's index.
-  ;; Given one of these lists, this function tries to add another splitindices to each cse.
-  (define (add-splitpoint v-alt-cost v-cidx v-pidx)
-    
-    ;; output vectors
-    (define vec-alt-cost (make-flvector num-points))
-    (define vec-cidx (make-vector num-points))
-    (define vec-pidx (make-vector num-points))
+  ;; Vectors are now filled with starting data. Beginning main loop of the
+  ;; regimes algorithm.
+ 
+  ;; Vectors used to determine if our current alt is better than out running
+  ;; best alt.
+  (define best-alt-idx (make-vector number-of-points))
+  (define best-alt-cost (make-vector number-of-points))
```

Comment:
```suggestion
  (define best-alt-cost (make-flvector number-of-points))
```
I think costs are always flonums

Reply:
So I had this originally but there wasn't a `flvector-fill!`. But just tested doing
`(set! best-alt-cost (make-flvector number-of-points +inf.0))` 
and it's faster on the hamming ci times.

Should of known on stack new vector is faster then pointer chases.